tv_apps/tv-epg/js/epg.js

```diff
@@ -45,10 +76,40 @@
     this.programDetailElement = document.getElementById('program-detail');
   };
 
+  /**
+   * Align timelineOffset and use it as x-axis translate reference point
+   */
+  proto._allTimelineAdded = function epg__allTimelineAdded() {
+    var timeDiff = this.initialTime - this.epgController.timelineOffset;
+    this.timelineElement.style.left =
+      (-timeDiff * this.columnWidth) + 'rem';
+    this.programListElement.style.left =
+      (-timeDiff * this.columnWidth) + 'rem';
+  };
+
+  /**
+   * Align channelOffset and use it as y-axis translate reference point
+   */
+  proto._allChannelFetched = function epg__allChannelFetched() {
+    this.visibleChannelOffset = this.epgController.channelOffset;
+    this.channelListElement.style.top =
+      (-this.visibleChannelOffset * this.rowHeight) + 'rem';
+    this.programListElement.style.top =
+      (-this.visibleChannelOffset * this.rowHeight) + 'rem';
+  };
+
   proto._onScanned = function epg__onScanned(stream) {
     this.videoElement.src = stream;
     this.epgController.fetchPrograms(
-      this.visibleTimeOffset - this.visibleTimeSize, 3 * this.visibleTimeSize);
+      this.visibleTimeOffset - this.visibleTimeSize, 3 * this.visibleTimeSize)
+      .then(function() {
```

Comment:
According to http://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html, we should add `catch` here to catch exceptions.